ospf6d/ospf6_top.c

```diff
@@ -1672,6 +1698,387 @@ DEFUN(show_ipv6_ospf6_route_type_detail, show_ipv6_ospf6_route_type_detail_cmd,
+}
+
+static int
+ospf6_show_summary_address(struct vty *vty, struct ospf6 *ospf6,
+			json_object *json,
+			bool uj, const char *detail)
+{
+	struct route_node *rn;
+	static const char header[] = "Summary-address       Metric-type     Metric     Tag         External_Rt_count\n";
+
+	if (!uj) {
+		vty_out(vty, "aggregation delay interval :%d(in seconds)\n\n",
+				ospf6->aggr_delay_interval);
+		vty_out(vty, "%s\n", header);
+	} else {
+		json_object_int_add(json, "aggregation delay interval",
+				ospf6->aggr_delay_interval);
+	}
+
+	for (rn = route_top(ospf6->rt_aggr_tbl); rn; rn = route_next(rn))
+		if (rn->info) {
+			struct ospf6_external_aggr_rt *aggr = rn->info;
+			json_object *json_aggr = NULL;
+			char buf[PREFIX2STR_BUFFER];
+
+			prefix2str(&aggr->p, buf, sizeof(buf));
+
+			if (uj) {
+
+				json_aggr = json_object_new_object();
+
+				json_object_object_add(json,
+							buf,
+							json_aggr);
+
+				json_object_string_add(json_aggr,
+						"Summary address",
+						buf);
+
+				json_object_string_add(
+					json_aggr, "Metric-type",
+					(aggr->mtype == DEFAULT_METRIC_TYPE)
+						? "E2"
+						: "E1");
+
+				json_object_int_add(json_aggr, "Metric",
+						   (aggr->metric != -1)
+						    ? aggr->metric
+						    : DEFAULT_DEFAULT_METRIC);
+
+				json_object_int_add(json_aggr, "Tag",
+						    aggr->tag);
+
+				json_object_int_add(json_aggr,
+						"External route count",
+						OSPF6_EXTERNAL_RT_COUNT(aggr));
+
+				if (OSPF6_EXTERNAL_RT_COUNT(aggr) && detail) {
+					json_object_int_add(json_aggr, "ID",
+							    aggr->id);
+					json_object_int_add(json_aggr, "Flags",
+							    aggr->aggrflags);
+					hash_walk(aggr->match_extnl_hash,
+					ospf6_print_json_external_routes_walkcb,
+							json_aggr);
+				}
+
+			} else {
+				vty_out(vty, "%-22s", buf);
+
+				(aggr->mtype == DEFAULT_METRIC_TYPE)
+					? vty_out(vty, "%-16s", "E2")
+					: vty_out(vty, "%-16s", "E1");
+				vty_out(vty, "%-11d", (aggr->metric != -1)
+						? aggr->metric
+						: DEFAULT_DEFAULT_METRIC);
+
+				vty_out(vty, "%-12u", aggr->tag);
+
+				vty_out(vty, "%-5ld\n",
+					OSPF6_EXTERNAL_RT_COUNT(aggr));
+
+				if (OSPF6_EXTERNAL_RT_COUNT(aggr) && detail) {
+					vty_out(vty,
+						"Matched External routes:\n");
+					hash_walk(aggr->match_extnl_hash,
+					ospf6_print_vty_external_routes_walkcb,
+								vty);
+					vty_out(vty, "\n");
+				}
+
+				vty_out(vty, "\n");
+			}
+		}
+
+	return CMD_SUCCESS;
+}
+
+DEFPY (show_ipv6_ospf6_external_aggregator,
+       show_ipv6_ospf6_external_aggregator_cmd,
+       "show ipv6 ospf6 summary-address [detail$detail] [json]",
+       SHOW_STR
+       IP6_STR
+       OSPF6_STR
+       "Show external summary addresses\n"
+       "detailed informtion\n"
+       JSON_STR)
+{
+	bool uj = use_json(argc, argv);
+	struct ospf6 *ospf6 = NULL;
+	json_object *json = NULL;
+
+	if (uj)
+		json = json_object_new_object();
+
+		/* Default Vrf */
+	ospf6 = ospf6_lookup_by_vrf_name(VRF_DEFAULT_NAME);
```

Comment:
Please make this command VRF-aware.

Reply:
Will be taken up as separate PR. Will be tracked using #9058

Reply:
I have added the VRF support in this PR.